src/storage/ring_buffer.rs

```diff
@@ -101,6 +119,67 @@ impl<'a, T: 'a> RingBuffer<'a, T> {
             Err(error) => Err(error)
         }
     }
+
+    /// Get capacity of the underlying storage.
```

Comment:
Weird to have `capacity()` but not `len()`. 

Style nit: these should go up after `full()` and `empty()`, where all the query methods live.